src/components/slider.js

```diff
@@ -0,0 +1,371 @@
+    const nextSlideIndex = this.state.currentSlideIndex - 1;
+    const actualNextSlide =
+      nextSlideIndex >= 0 ? nextSlideIndex : this.slideCount - 1;
+    this.goTo(actualNextSlide, PREVIOUS);
+  };
+
+  next = () => {
+    if (!this.canGoNext()) return;
+    const nextSlideIndex = (this.state.currentSlideIndex + 1) % this.slideCount;
+    this.goTo(nextSlideIndex, NEXT);
+  };
+
+  getSlideClass = index => {
+    const { currentSlideIndex, animation } = this.state;
+    const classNames = this.getClassNames();
+    const lastSlideIndex = this.slideCount - 1;
+    if (index === currentSlideIndex) {
+      if (animation) return `${classNames.animateOut} ${classNames[animation]}`;
+      return classNames.current;
+    }
+    if (this.slideCount === 2) {
+      if (animation) return `${classNames.animateIn} ${classNames[animation]}`;
+      return index < currentSlideIndex ? classNames.previous : classNames.next;
+    }
+    if (
+      index === currentSlideIndex - 1 ||
+      (currentSlideIndex === 0 && index === lastSlideIndex)
+    ) {
+      if (animation === PREVIOUS)
+        return `${classNames.animateIn} ${classNames.previous}`;
+      if (animation === NEXT) return classNames.hidden;
+      return classNames.previous;
+    }
+    if (
+      index === currentSlideIndex + 1 ||
+      (index === 0 && currentSlideIndex === lastSlideIndex)
+    ) {
+      if (animation === NEXT)
+        return `${classNames.animateIn} ${classNames.next}`;
+      if (animation === PREVIOUS) return classNames.hidden;
+      return classNames.next;
+    }
+    return classNames.hidden;
+  };
+
+  /* eslint-disable lines-between-class-members */
+  isSwiping = false;
+  sliderRef;
+  pageStartPosition;
+
+  currentElement;
+  currentElementStartPosition;
+  currentElementPosition;
+  previousElement;
+  previousElementStartPosition;
+  previousElementPosition;
+  nextElement;
+  nextElementStartPosition;
+  nextElementPosition;
+  /* eslint-enable lines-between-class-members */
+
+  handleTouchStart = e => {
+    if (this.isDisabled()) return;
+    this.stopAutoplay();
+    const { current, previous, next } = this.getClassNames();
+    const touch = e.touches[0];
+    this.isSwiping = true;
+    this.pageStartPosition = touch[this.swipeEventProperty];
+    /* eslint-disable prefer-destructuring */
+    this.currentElement = this.sliderRef.getElementsByClassName(current)[0];
+    this.previousElement = this.sliderRef.getElementsByClassName(previous)[0];
+    this.nextElement = this.sliderRef.getElementsByClassName(next)[0];
+    /* eslint-enable prefer-destructuring */
+    const touchDelta = this.currentElement.getBoundingClientRect()[
+      this.swipeProperty
+    ];
+    this.currentElementStartPosition = 0;
+    this.currentElementPosition = 0;
+    this.currentElement.style.transition = "none";
+    if (this.previousElement) {
+      this.previousElement.style.transition = "none";
+      this.previousElement.style.visibility = "visible";
```

Comment:
I would recommend doing this via CSS style rather than a programmatic approach just for maintainability.